### PR TITLE
Add Wiki Sidebar, Use Gatsby Generated Wiki

### PR DESCRIPTION
**Pull request by SaintPeter**
Add a sidebar identical to the chat and map sidebars that hosts an iframe with the Gatsby generated Wiki pages.

Also adds a full-page version of the Wiki via an iframe.

Tested Locally.

### CODE REVIEW

**Comment by SaintPeter** [comment]
@hallaathrad Akira reports that the dropdown in the sidebar is not properly changing pages.  I thought we had tested that?

**Comment by SaintPeter** [comment]
I have resolved the issue with the drop-down not working.  This should be good to go.

Please QA.

**Comment by jamigibbs** [comment]
Just stopping by to mention that you may need to add webkit support for iOS devices. I was working through a mobile issue reported for the map sidebar that might be relevant here too: #6227

File: server/views/wiki/show.jade [hunk 1]
```diff
@@ -0,0 +1,7 @@
+extends ../layout-wide
+block content
+    iframe#wikiFrame(frameborder='no')
+    script.
+        var lang = window.location.toString().match(/\/\w{2}\//);
+        lang = (lang) ? lang[0] : '/en/';
+        $('#wikiFrame').attr('src','http://freecodecamp.github.io/wiki' + lang);
```
Comment (BerkeleyTrue) [inline server/views/wiki/show.jade]: Looks like we won't need to proxy the wiki using nginx? Also, how does this work in local dev?

File: server/middlewares/csp.js [hunk 2]
```diff
@@ -65,7 +65,9 @@ export default function csp() {
         '*.vimeo.com',
         '*.twitter.com',
         '*.ghbtns.com',
-        '*.freecatphotoapp.com'
+        '*.freecatphotoapp.com',
+        'freecodecamp.github.io',
+        'localhost:8000'        
       ].concat(trusted)
```
Comment (BerkeleyTrue) [inline server/middlewares/csp.js]: What is this url for?

File: client/main.js [hunk 3]
```diff
@@ -376,6 +376,37 @@ $(document).ready(function() {
...
+  function showWiki() {
+    if (!main.isWikiAsideLoad) {
+      var lang = window.location.toString().match(/\/\w{2}\//);
+      lang = (lang) ? lang[0] : '/en/';
+      var wikiURL = 'http://freecodecamp.github.io/wiki' + lang;
+      var wikiAside = $('<iframe>');
+      wikiAside.attr('src', wikiURL);
+      $('.wiki-aside').append(wikiAside);
+      main.isWikiAsideLoad = true;
+    }
```
Comment (BerkeleyTrue) [inline client/main.js]: This process will continuously add iframes to the DOM is the user opens and close the aside.

File: server/boot/t-wiki.js [hunk 4]
```diff
@@ -0,0 +1,10 @@
+module.exports = function(app) {
+  var router = app.loopback.Router();
+  router.get('/wiki', showWiki);
+
```
Comment (BerkeleyTrue) [inline server/boot/t-wiki.js]: We can change this to '/wiki/*' and it will catch all the routes.

**Comment by QuincyLarson** [comment]
One big problem: on OSX in Chrome 48 I am unable to scroll.

File: client/main.js (hunk 3 above)
Comment (SaintPeter) [inline client/main.js]: See the `if(!main.isWikiAsideLoad)` and the `main.isWikiAsideLoad = true`?  That should cause only one iframe to be created.  This is literally identical code to the map toggle which we are using - I cut, pasted, and did a search and replace for the word `map`.

File: server/views/wiki/show.jade (hunk 1 above)
Comment (SaintPeter) [inline server/views/wiki/show.jade]: This works for local dev, but you have to push changes to the "live" ga-pages wiki to test.

**Comment by SaintPeter** [comment]
Updated:  
Added routes for `/wiki/* and `/wiki`
Added fix for OSX scrolling

**Comment by QuincyLarson** [comment]
This works great now! 
One note: 
All links should probably be "/article-title" instead of "freecodecamp.com/article-title" but this won't matter once we push to production. Are you all ready for me to merge this? I think we can deploy it tonight.

**Comment by hallaathrad** [comment]
:godmode: go go go

**Comment by SaintPeter** [comment]
I think this is ready to merge.  We're going to have to figure out how to deal with the links generally.  I've got that as part of the conversaion script, but the new links don't work quite properly.

**Pull request merged**